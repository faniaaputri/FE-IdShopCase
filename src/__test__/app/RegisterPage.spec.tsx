import RegisterPage from "@/app/(auth)/register/page";
import Home from "@/app/page";
import { render } from "@testing-library/react";

describe("Register Page", () => {
  it("should render", () => {
    const page = render(<RegisterPage />);
    expect(page).toMatchSnapshot();
  });
});
