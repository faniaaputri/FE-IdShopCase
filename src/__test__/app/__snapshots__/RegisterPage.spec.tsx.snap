// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Register Page should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="flex flex-col gap-5 mt-5"
      >
        <div>
          <label
            for="username"
          >
            Name
          </label>
          <input
            class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="input"
            id="username"
            placeholder="Username"
            type="text"
          />
        </div>
        <div>
          <label
            for="email"
          >
            Email
          </label>
          <input
            class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="input"
            id="email"
            placeholder="Email"
            type="email"
          />
        </div>
        <div>
          <label
            for="phone"
          >
            No Handphone
          </label>
          <input
            class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="input"
            id="phone"
            placeholder="Phone"
            type="text"
          />
        </div>
        <div>
          <label
            for="password"
          >
            Password
          </label>
          <div
            class="relative w-full"
          >
            <input
              class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive pr-10"
              data-slot="input"
              id="password"
              placeholder="Password"
              type="password"
            />
            <div
              class="absolute right-2 top-1/2 -translate-y-1/2 cursor-pointer "
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-eye-closed text-ring"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m15 18-.722-3.25"
                />
                <path
                  d="M2 8a10.645 10.645 0 0 0 20 0"
                />
                <path
                  d="m20 15-1.726-2.05"
                />
                <path
                  d="m4 15 1.726-2.05"
                />
                <path
                  d="m9 18 .722-3.25"
                />
              </svg>
            </div>
          </div>
        </div>
        <div>
          <label
            for="confirm-password"
          >
            Confirm Password
          </label>
          <div
            class="relative w-full"
          >
            <input
              class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive pr-10"
              data-slot="input"
              id="confirm-password"
              placeholder="Confirm Password"
              type="password"
            />
            <div
              class="absolute right-2 top-1/2 -translate-y-1/2 cursor-pointer "
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-eye-closed text-ring"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m15 18-.722-3.25"
                />
                <path
                  d="M2 8a10.645 10.645 0 0 0 20 0"
                />
                <path
                  d="m20 15-1.726-2.05"
                />
                <path
                  d="m4 15 1.726-2.05"
                />
                <path
                  d="m9 18 .722-3.25"
                />
              </svg>
            </div>
          </div>
        </div>
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 font-bold"
          data-slot="button"
        >
          Daftar
        </button>
        <div
          class="flex justify-center gap-1"
        >
          <p>
            Kamu sudah punya akun?
          </p>
          <a
            class="text-primary font-bold"
            href="/login"
          >
            Masuk
          </a>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="flex flex-col gap-5 mt-5"
    >
      <div>
        <label
          for="username"
        >
          Name
        </label>
        <input
          class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="input"
          id="username"
          placeholder="Username"
          type="text"
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email
        </label>
        <input
          class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="input"
          id="email"
          placeholder="Email"
          type="email"
        />
      </div>
      <div>
        <label
          for="phone"
        >
          No Handphone
        </label>
        <input
          class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="input"
          id="phone"
          placeholder="Phone"
          type="text"
        />
      </div>
      <div>
        <label
          for="password"
        >
          Password
        </label>
        <div
          class="relative w-full"
        >
          <input
            class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive pr-10"
            data-slot="input"
            id="password"
            placeholder="Password"
            type="password"
          />
          <div
            class="absolute right-2 top-1/2 -translate-y-1/2 cursor-pointer "
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-eye-closed text-ring"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15 18-.722-3.25"
              />
              <path
                d="M2 8a10.645 10.645 0 0 0 20 0"
              />
              <path
                d="m20 15-1.726-2.05"
              />
              <path
                d="m4 15 1.726-2.05"
              />
              <path
                d="m9 18 .722-3.25"
              />
            </svg>
          </div>
        </div>
      </div>
      <div>
        <label
          for="confirm-password"
        >
          Confirm Password
        </label>
        <div
          class="relative w-full"
        >
          <input
            class="file:text-foreground placeholder:text-muted-foreground/70 selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-11 w-full min-w-0 rounded-sm border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive pr-10"
            data-slot="input"
            id="confirm-password"
            placeholder="Confirm Password"
            type="password"
          />
          <div
            class="absolute right-2 top-1/2 -translate-y-1/2 cursor-pointer "
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-eye-closed text-ring"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15 18-.722-3.25"
              />
              <path
                d="M2 8a10.645 10.645 0 0 0 20 0"
              />
              <path
                d="m20 15-1.726-2.05"
              />
              <path
                d="m4 15 1.726-2.05"
              />
              <path
                d="m9 18 .722-3.25"
              />
            </svg>
          </div>
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 font-bold"
        data-slot="button"
      >
        Daftar
      </button>
      <div
        class="flex justify-center gap-1"
      >
        <p>
          Kamu sudah punya akun?
        </p>
        <a
          class="text-primary font-bold"
          href="/login"
        >
          Masuk
        </a>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
